src/app/questionnaire-form-filler/questionnaire-form-filler.component.ts

```diff
@@ -71,30 +99,31 @@ export class QuestionnaireFormFillerComponent implements OnInit {
         body: bundle,
       });
     }
-    await this.saveQuestionnaireResponse('completed');
+    await this.saveQuestionnaireResponse(questionnaireResponse, 'completed');
     await this.router.navigateByUrl('/');
   }
 
-  onSaveAsDraft() {
-    return this.saveQuestionnaireResponse().then(() =>
-      this.router.navigateByUrl('/')
-    );
+  onSaveAsDraft(initialQuestionnaireResponse?: fhir.r4.QuestionnaireResponse) {
```

Comment:
is this really the initial qr?

Reply:
Yes. We need it here to check if we should update it or create a new QuestionnaireResponse.